summoner-tui/src/Summoner/Tui/Validation.hs

```diff
@@ -105,16 +108,22 @@ showFormError = \case
 -- | Returns list of all invalid fields according to the error.
 errorToInvalidFields :: FormError -> NonEmpty SummonForm
 errorToInvalidFields = \case
-    EmptyFields fields -> fields
-    OneWord fields     -> fields
-    ProjectExist       -> one ProjectName
-    CabalOrStack       -> CabalField :| [StackField]
-    LibOrExe           -> Lib :| [Exe]
+    EmptyFields fields  -> fields
+    OneWord fields      -> fields
+    ProjectExist        -> one ProjectName
+    CabalOrStack        -> CabalField :| [StackField]
+    LibOrExe            -> Lib :| [Exe]
+    PreludePackageValid -> one CustomPreludeName
 
 -- | Takes boolean value and error and returns error if predicate 'True'.
 toError :: Bool -> e -> Validation (NonEmpty e) ()
 toError p e = if p then Failure (one e) else Success ()
 
+-- | Decides whether the given text is a valid package name. Spec is here:
+-- https://www.haskell.org/cabal/users-guide/developing-packages.html#package-names-and-versions
+packageNameValid :: Text -> Bool
+packageNameValid = T.all (\c -> c == '-' || C.isAlphaNum c)
```

Comment:
I believe this function belongs to the core package, so we would be able to reuse it for CLI as well.